File: cordapp-cpk2/src/test/resources/cordapp-embedded-overrides/src/main/java/com/example/embeddedoverrides/ExampleContract.java
package com.example.embeddedoverrides;

import net.corda.v5.ledger.contracts.Contract;
import net.corda.v5.ledger.transactions.LedgerTransaction;
import org.jetbrains.annotations.NotNull;
import org.slf4j.LoggerFactory;
import org.slf4j.Logger;

public class ExampleContract implements Contract {
    private final Logger logger = LoggerFactory.getLogger(ExampleContract.class);

    @Override
    public void verify(@NotNull LedgerTransaction ltx) {
        logger.info("Verify transaction: {}", ltx);
    }
}

File: cordapp-cpk/src/test/resources/configure-corda-metadata/cordapp/src/main/kotlin/com/example/metadata/custom/ExampleContract.kt
@file:Suppress("unused")
package com.example.metadata.custom

import net.corda.core.identity.AbstractParty
import net.corda.v5.ledger.contracts.Contract
import net.corda.v5.ledger.contracts.ContractState
import net.corda.v5.ledger.transactions.LedgerTransaction
import java.util.Collections.unmodifiableList

class ExampleContract : Contract {
    override fun verify(ltx: LedgerTransaction) {
        @Suppress("ObjectLiteralToLambda")
        val innerContract = object : Contract {
            override fun verify(ltx: LedgerTransaction) {
                println(ltx.toString())
            }
        }
        innerContract.verify(ltx)

        val lambdaContract = Contract { l -> println(l.toString()) }
        lambdaContract.verify(ltx)
    }

    class NestedContract : Contract {
        override fun verify(ltx: LedgerTransaction) = throw UnsupportedOperationException(this::class.java.name)
    }

    inner class InnerContract : Contract {
        override fun verify(ltx: LedgerTransaction) = throw UnsupportedOperationException(this::class.java.name)
    }
}

class ExampleState(val issuer: AbstractParty) : ContractState {
    override val participants: List<AbstractParty> = unmodifiableList(listOf(issuer))
}
